aiida/orm/utils/managers.py

```diff
@@ -80,8 +81,10 @@ def __getattr__(self, name):
         """
         try:
             return self._get_node_by_link_label(label=name)
-        except KeyError:
-            raise AttributeError("Node '{}' does not have an input with link '{}'".format(self._node.pk, name))
+        except exc.NotExistent:
+            raise exc.NotExistentAttributeError(
```

Comment:
Was the idea that we are going to deprecate this at some point? I.e. for AiIDA 2.0 this will just raise an `AttributeError`? Not sure how to do that, because I don't think we can inspect when people catch a `NotExistent` exception. But if we don't we will always have to keep this composed exception just for this case. Same goes for the `NotExistentKeyError`.

If we have to keep it, can we add a quick comment here, where it is being raised why we need this special exception?

Reply:
I think we don't need to deprecate, and that using this compound exception is safe to keep also in future versions. I have added more comments to explain the reason for this. Now I also rebased, and squashed all in 1 commit, so it's ready to merge (unless I broke tests in the meantime ;-) ).